cost/aws/old_snapshots/aws_delete_old_snapshots.pt

```diff
@@ -0,0 +1,261 @@
+    "verb": "GET",
+    "path": "/",
+    "headers": {
+      "User-Agent": "RS Policies",
+      "Content-Type": "text/xml"
+    },
+    "query_params": {
+      'Action': 'DescribeSnapshots',
+      'Version': '2016-11-15',
+      'Filter.1.Name': 'owner-id',
+      'Filter.1.Value.1': ds_get_caller_identity[0]['account']
+    }
+  }
+EOS
+end
+
+script "js_filter_old_snapshots", type: "javascript" do
+  result "results"
+  parameters "snapshot_age", "ds_describe_snapshots", "param_exclude_tags"
+  code <<-EOS
+      var param_exclude_tags_lower=[];
+      for(var j=0;j<param_exclude_tags.length;j++){
+        param_exclude_tags_lower[j] = param_exclude_tags[j].toString().toLowerCase();
+      }
+  var results = [];
+  var start_date = new Date(new Date().setDate(new Date().getDate() - snapshot_age));
+  var start_date_time=start_date.getTime()
+  console.log(start_date_time);
+   _.each(ds_describe_snapshots,function(snapshot) {
+      var snapshot_time =  Date.parse(snapshot['startTime']);
+       console.log(snapshot_time);
+        // Check, if the tag present in entered param_exclude_tags, ignore the volume if the tag matches/present.
+        var tags = snapshot['tags'];
+        var isTagMatched = false;
+        var tagKeyValue = "";
+        for(var k=0; k < tags.length; k++){
+        tag = tags[k];
+        if((param_exclude_tags_lower.indexOf((tag['tagKey']).toLowerCase()) !== -1) || (param_exclude_tags_lower.indexOf((tag['tagKey']+'='+tag['tagValue']).toLowerCase()) !== -1)){
+          isTagMatched = true;
+        }
+
+        // Constructing tags with comma separated to display in detail_template
+        if((tag['tagValue']).length > 0){
+          tagKeyValue = tagKeyValue+" , "+tag['tagKey']+'='+tag['tagValue'];
+        }else{
+          tagKeyValue = tagKeyValue+" , "+tag['tagKey'];
+        }
+      }
+      if(snapshot_time < start_date_time && !isTagMatched){
+           results.push({
+                snapshotId : snapshot['snapshotId'],
+                region : snapshot['region'],
+                tagKeyValue : (tagKeyValue.slice(2)),
+                status : snapshot['status'],
+				startTime : snapshot['startTime'],
+                ownerAlias : snapshot['ownerAlias']
+              })
+      }
+   });
+
+EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_utilization" do
+  validate $ds_filter_old_snapshots do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} rows containing AWS old snapshot data"
+    detail_template <<-EOS
+# AWS Old Snapshot
+| Region | SnapshotId | Status | Created | Description | Tags |
+| ------ | ---------- | ------ | ------- | ----------- | ---- |
+{{ range data -}}
+| {{.region}} | {{.snapshotId}} | {{.status}} |  {{.startTime }} | {{ .description }} | {{ .tagKeyValue }} |  
```

Comment:
Is status needed.  We should probably only be filtering on any snapshot that is completed.  anything order than a few minutes/hrs would be complete.  consider removing this field and filtering by status==complete

Reply:
Instead of using Created, change to Age and consider sorting by region and age and moving Age to be the 2nd column.

Reply:
I have removed the status field and also filtering by status == completed. 

I have added days old field and I haven't removed created date field because of the user what to know when the snapshot has been created. If you want I'll remove the created date field